membership-attribute-service/app/models/AccessClaims.scala

```diff
@@ -0,0 +1,41 @@
+package models
+
+import com.gu.identity.auth.{ClaimsParser, DefaultClaimsParser, UnparsedClaims, UserClaims, ValidationError}
+import com.gu.identity.model.User
+
+/** Claims that are used to determine which resources the user is authorised to access.
+  *
+  * @param primaryEmailAddress
+  *   primary email address
+  * @param identityId
+  *   Identity ID
+  * @param username
+  *   User name, a synonym for display name. Also used to determine if this is a test user.
+  * @param userEmailValidated
+  *   true iff the user has validated their email address
+  */
+case class AccessClaims(
+                         primaryEmailAddress: String,
+                         identityId: String,
+                         username: Option[String] = None,
+                         userEmailValidated: Option[Boolean] = None,
+) extends UserClaims
```

Comment:
It might be that `userEmailValidated` in needed in multiple places and it makes sense to it to the default access claims defined in the library.  I added them here because 1) it allows me to demonstrate how to use custom claims defined outside the library and 2) it's a mutable field that could get out of date so maybe we should avoid getting it from the claims as much as possible.

as a semi related comment we should really think about the consequences of reading a stale field in each case. It might make sense to force the user to renew their access token after updating their email or setting their username or things like that. Although it probably won't be possible or at least easy to do across devices